### PR TITLE
Update the download all files location

### PR DESCRIPTION
**Pull request by MichaelRBond**
On large downloads, the /tmp directory was filling up and causing the download to be 0 bytes. This addresses that issue. 

This is to address the issue in ticket: https://systems.lib.wvu.edu/helpdesk/viewTicket/?id=14773

THis has been tested on MFCS dev with record P31 and P3001. P31 worked previously and continues to work with this update. P3001 was too large to download previously, and now works correctly.

### CODE REVIEW

File: public_html/dataView/allfiles.php [hunk 1]
```diff
@@ -41,6 +41,8 @@
 	}
 
 	//determine the field name
+	// FIXME: this will only grab the LAST file field, if a form has Multiple
```
Comment (ddavisgraphics) [inline public_html/dataView/allfiles.php]: This appears incomplete, if there is a FIXME in the comments your acknowledging that the code doesn't work properly.  These are the things I want to avoid merging into the codebase.

File: public_html/dataView/allfiles.php [hunk 2]
```diff
@@ -86,31 +86,12 @@
 		header("Cache-Control: public");
 		header("Content-Description: File Transfer");
 		header("Content-type: application/zip");
-		header("Content-Disposition: attachment; filename=\"".$object['idno'].".".$type."\"");
+		header(sprintf('Content-Disposition: attachment; filename="%s.%s"',$object['idno'],$type));
```
Comment (ddavisgraphics) [inline public_html/dataView/allfiles.php]: The previous version included slashes for the directory, is this not necessary?

File: public_html/dataView/allfiles.php [hunk 3]
```diff
@@ -53,13 +55,11 @@
...
-			$files[] = sprintf("%s",$file['name']);
-
+			$files[] = $file['name'];
 		}
 
 		$files           = implode(" ",$files);
-		$destinationFile = sys_get_temp_dir()."/".time().".".$type;
+		$destinationFile = sprintf("%s/%s.%s",mfcs::config('mfcsDownloadAllDir'),time(),$type);
 
 		if ($type == "zip") {
```
Comment (ddavisgraphics) [inline public_html/dataView/allfiles.php]: I'm curious, would it be better to do something similar to the export scripts where you use symbolic linking, follow the links, and then zip the file following the symbolic links.

File: public_html/dataView/allfiles.php (hunk 2 above)
Comment (MichaelRBond) [inline public_html/dataView/allfiles.php]: the previous version included slashed escaping the quotes

File: public_html/dataView/allfiles.php (hunk 3 above)
Comment (MichaelRBond) [inline public_html/dataView/allfiles.php]: That might make it more performant (likely would), but that should be addressed in a separate pull request. 

Doing that would not address the problem that the temporary directory isn't configurable, which this does.

File: public_html/dataView/allfiles.php (hunk 1 above)
Comment (MichaelRBond) [inline public_html/dataView/allfiles.php]: there is absolutely nothing wrong with fixme and todos in code. They are reminders and notes. This code does not introduce a new issue. it is noting an existing issue that was discovered while reviewing code.

that existing issue is in no way related to the current problem. It should be addressed at a later date.

File: public_html/dataView/allfiles.php (hunk 2 above)
Comment (ddavisgraphics) [inline public_html/dataView/allfiles.php]: Duh... I see that now.  Sorry, I was thinking filesystem while looking at it.
